code/modules/mob/inventory.dm

```diff
@@ -205,8 +205,26 @@
 /mob/proc/put_in_hands(obj/item/I, del_on_fail = FALSE)
 	if(!I)
 		return FALSE
+
+	// If the item is a stack and we're already holding a stack then merge
+	if (istype(I, /obj/item/stack))
+		var/obj/item/stack/I_stack = I
+		var/obj/item/stack/active_stack = get_active_held_item()
+
+		if (active_stack && istype(I_stack, active_stack.merge_type))

```

Comment:
you can assign incorrect types to typed vars just fine and this will runtime, you need to do:
`if(istype(active_stack) && ...` instead